ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -519,60 +530,73 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
     }
 
     async getPreviewList() {
-      this.setState({previewList: [], previewDataLoading: true});
       const {namespace, id} = this.props.workspace;
-      const request = {
-        name: '',
-        description: '',
-        conceptSetIds: this.state.selectedConceptSetIds,
-        includesAllParticipants: this.state.includesAllParticipants,
-        cohortIds: this.state.selectedCohortIds,
-        prePackagedConceptSet: this.getPrePackagedConceptSet(),
-        values: this.state.selectedValues
-      };
-      try {
-        const dataSetPreviewResp = await dataSetApi().previewQuery(namespace, id, request);
-        this.setState({
-          previewList: dataSetPreviewResp.domainValue,
-          selectedPreviewDomain: dataSetPreviewResp.domainValue[0].domain
-        });
-      } catch (ex) {
-        const exceptionResponse = await ex.json() as unknown as ErrorResponse;
-        let errorText: string;
-        switch (exceptionResponse.statusCode) {
-          case 400:
-            if (exceptionResponse.message ===
-              'Data Sets must include at least one cohort and concept.') {
-              errorText = exceptionResponse.message;
-            } else if (exceptionResponse.message ===
-              'Concept Sets must contain at least one concept') {
-              errorText = 'One or more of your concept sets has no concepts. ' +
-                'Please check your concept sets to ensure all concept sets have concepts.';
-            }
-            break;
-          case 404:
-            if (exceptionResponse.message.startsWith(
-              'Not Found: No Cohort definition matching cohortId')) {
-              errorText = 'Error with one or more cohorts in the data set. ' +
-                'Please submit a bug using the contact support button';
-            }
-            break;
-          case 504:
-            if (exceptionResponse.message ===
-              'Timeout while querying the CDR to pull preview information.') {
-              errorText = 'Query to load data from the All of Us Database timed out. ' +
-                'Please either try again or export data set to a notebook to try there';
-            }
-            break;
-          default:
-            errorText = 'An unexpected error has occurred. ' +
-              'Please submit a bug using the contact support button';
+      const domains = fp.uniq(this.state.selectedValues.map(domainValue => domainValue.domain));
```

Comment:
The `selectedValues` object is generated, although we could create a wrapper. I have no strong objections to doing that, but it would add a layer of abstraction around the DomainValuePair object.

Reply:
`previewList = new Map(this.state.selectedValues.map(selection => [selection.domain, {isLoading: true, values: []}]))`

do you know if this would work? Based off of this - https://stackoverflow.com/questions/26264956/convert-object-array-to-hash-map-indexed-by-an-attribute-value-of-the-object

Reply:
So, I tried this, and I am getting some truly obnoxious typing errors:
```ERROR in src/app/pages/data/data-set/dataset-page.tsx(549,17): error TS2345: Argument of type '(Domain | DataSetPreviewInfo)[][]' is not assignable to parameter of type 'ReadonlyArray<[Domain, DataSetPreviewInfo]>'.
  Type '(Domain | DataSetPreviewInfo)[]' is not assignable to type '[Domain, DataSetPreviewInfo]'.
    Property '0' is missing in type '(Domain | DataSetPreviewInfo)[]'```

Reply:
Its not entirely clear to me why map did it that way, where when I use the explicit constructor for `new Array` it doesn't do it that way, but spits out some other errors.

Reply:
this would be a useful mob session depending on time. I don't quite follow what we're doing.

Reply:
Got it. I am happy to do a knowledge sharing session at some point.